autoload/go/path.vim

```diff
@@ -139,11 +139,6 @@ function! go#path#CheckBinPath(binpath)
     " remove whitespaces if user applied something like 'goimports   '
     let binpath = substitute(a:binpath, '^\s*\(.\{-}\)\s*$', '\1', '')
 
-    " if it's in PATH just return it
-    if executable(binpath) 
-        return binpath
-    endif
-
```

Comment:
Yeah, maybe this isn't the best way to do this, but, without that it always uses what's present in `$PATH`, regardless of the ordering changes. Maybe we should move the `let $PATH` line up above this block?